src/NumberTheory/QuadFormAndIsom/torsion_quadratic_module_with_isometry.jl

```diff
@@ -0,0 +1,526 @@
+julia> Tf = torsion_quadratic_module_with_isometry(QQ[-1//60;], ZZ[11;])
+Finite quadratic module of order 60
+  with 1 generator
+  with isometry given by
+  [11]
+
+julia> order_of_isometry(Tf)
+2
+```
+"""
+@attr Int function order_of_isometry(Tf::TorQuadModuleWithIsom)
+  T = underlying_module(Tf)
+  f = isometry(Tf)
+  C = Oscar._orthogonal_group(T, [f]; check=false)
+  return Int.(order(C))
+end
+
+###############################################################################
+#
+#  Constructors
+#
+###############################################################################
+
+@doc raw"""
+    torsion_quadratic_module_with_isometry(
+      T::TorQuadModule,
+      [f::U];
+      check::Bool=true,
+    ) -> TorQuadModuleWithIsom
+
+Given a torsion quadratic module ``T`` and given ``f`` being either:
+  * an isometry of ``T``
+    (`U <: AutomorphismGroupElem{TorQuadModule}`),
+  * an abelian group endomorphism of ``T``
+    (`U <: Union{TorQuadModuleMap, FinGenAbGroupHom, ZZMatrix}`),
+  * or an isometry of the cover ``L`` of ``T``
+    (`U <: MatrixGroupElem{QQFieldElem, QQMatrix}`)
+return the associated torsion quadratic module with isometry ``(T, f')``,
+where ``f' <: TorQuadModuleMap`` is the isometry of ``T`` defined by ``f``.
+
+If ``f`` is not supplied, it is set to be the identity map of ``T`` by
+default.
+
+If `check` is set to `true`, the function checks whether ``f`` indeed defines
+an isometry of ``T``.
+"""
+torsion_quadratic_module_with_isometry(::TorQuadModule, ::TorQuadModuleMap)
+
+function torsion_quadratic_module_with_isometry(
+  T::TorQuadModule,
+  f::TorQuadModuleMap;
+  check::Bool=true,
+)
+  @req is_finite(abelian_group(T)) "Only available for finite modules"
+  if check
+    @req domain(f) === codomain(f) === T "Wrong domain or codomain"
+    @req Hecke.is_isometry(f) "Not an isometry"
+  end
+
+  return TorQuadModuleWithIsom(T, f)
+end
+
+function torsion_quadratic_module_with_isometry(T::TorQuadModule)
+  @req is_finite(abelian_group(T)) "Only available for finite modules"
+  f = id_hom(T)
+  return TorQuadModuleWithIsom(T, f)
+end
+
+function torsion_quadratic_module_with_isometry(
+  T::TorQuadModule,
+  f::AutomorphismGroupElem{TorQuadModule};
+)
+  @req is_finite(abelian_group(T)) "Only available for finite modules"
+  @req domain(f) === T "Wrong domain"
+  return torsion_quadratic_module_with_isometry(T, hom(f); check=false)
+end
+
+function torsion_quadratic_module_with_isometry(
+  T::TorQuadModule,
+  m::ZZMatrix;
+  check::Bool=true,
+)
+  @req is_finite(abelian_group(T)) "Only available for finite modules"
+  f = hom(T, T, m; check)
+  return torsion_quadratic_module_with_isometry(T, f; check)
+end
+
+function torsion_quadratic_module_with_isometry(
+  T::TorQuadModule,
+  g::FinGenAbGroupHom;
+  check::Bool=true,
+)
+  @req is_finite(abelian_group(T)) "Only available for finite modules"
+  if check
+    @req domain(g) === codomain(g) === abelian_group(T) "Wrong domain or codomain"
+  end
```

Comment:
This is so cheap, that I do not see a point in turning off this check.
```suggestion
  @req domain(g) === codomain(g) === abelian_group(T) "Wrong domain or codomain"
```